File: lucky_num.java
import java.io.*;
class Main
{
	public static int counter = 2;
	static boolean isLucky(int n)
	{
		if(counter > n)
			return true;
		if(n%counter == 0)
			return false;	
		int next_position = n - (n/counter);	
		counter++;
		return isLucky(next_position);
	}
 public static void main (String[] args)
	{	
    int x = 5;
		if( isLucky(x) )
			System.out.println(x+" is a lucky no.");
		else
			System.out.println(x+" is not a lucky no.");
	}
}

File: lucky_num.py
def isLucky(n):
	if isLucky.counter > n:
		return 1
	if n % isLucky.counter == 0:
		return 0
	next_position = n - (n/isLucky.counter)	
	isLucky.counter = isLucky.counter + 1
	return isLucky(next_position)
isLucky.counter = 2 
x = 5
if isLucky(x):
	print x,"is a Lucky number"
else:
	print x,"is not a Lucky number"
